lib/firebase.ts

```diff
@@ -29,7 +29,62 @@ export function docToJSON(doc: DocumentSnapshot) {
     }
 }
 
+export async function fetchMasterList(urlUsername: string) {
+    const ref = collection(getFirestore(), 'users');
+    const userInfoQuery = query(
+        // @ts-ignore
+        ref,
+        where('username', "==", urlUsername)
+    )
+
+    // @ts-ignore
+    const userInfo = docToJSON((await getDocs(userInfoQuery)).docs[0]);
+
+    console.log("DB READ!")
+
+    return userInfo.listedMovies;
+}
+
+
+export async function fetchUserInfo(userId: string) {
+    console.log("DB READ!")
+    return getDoc(doc(firestore,"users", userId))
+
+}
+export async function deleteMovieFromList(id: number, list: string, userId: string) {
+    console.log("DB READ!")
+    fetchUserInfo(userId).then((res) => {
+        var listedMovies: ListedMovie[] = res.get("listedMovies")
+
+        listedMovies[listedMovies.findIndex(x => x.movieId == id)].lists = listedMovies[listedMovies.findIndex(x => x.movieId == id)].lists.filter(x => x !== list)
+        let data = {
+            listedMovies: listedMovies
+        }
+        setDoc(doc(firestore,"users", userId), data, { merge:true })
+
+    })
+
+}
+
+export async function fetchMovieDetailsForList(movieIds: [number]) {
+    console.log("DB READ!")
+    const ref = collection(getFirestore(), 'movies');
+    const searchQuery = query(
+        // @ts-ignore
+        ref,
+        where('id', "in", movieIds),
+        orderBy("title")
+    )
+
+
+
+    // console.table(userInfo);
+    // @ts-ignore
+    return (await getDocs(searchQuery)).docs.map(docToJSON);
+}
+
 export async function getMovieById(id : number) {
+    console.log("DB READ!")
```

Comment:
Be careful with the console.logs. These can potentially expose information to the user, and will cause slowdowns in your code.

Reply:
I am aware of that but we have had many major issues where the DB is called way too often so I will be leaving that in for now so we do not get kicked out of the DB again